src/uct/cuda/cuda_ipc/cuda_ipc_ep.h

```diff
@@ -32,4 +47,29 @@ ucs_status_t uct_cuda_ipc_ep_put_zcopy(uct_ep_h tl_ep,
                                        const uct_iov_t *iov, size_t iovcnt,
                                        uint64_t remote_addr, uct_rkey_t rkey,
                                        uct_completion_t *comp);
+
+static inline uint64_t uct_cuda_ipc_rem_seg_hash(uct_cuda_ipc_rem_seg_t *seg)
+{
+    uint64_t hash_val = 7;
+    int i;
+
+    for (i = 0; i < sizeof(seg->ph); i++) {
+        hash_val = hash_val*31 + seg->ph.reserved[i];
+    }
+
+    return (uint64_t) (hash_val % UCT_CUDA_IPC_HASH_SIZE);
+}
+
+static inline int64_t uct_cuda_ipc_rem_seg_compare(uct_cuda_ipc_rem_seg_t *sg1,
```

Comment:
looks like no need for casting, `int` would be enough.
BTW, previously it was found that KHASH has better performance/scalability in many cases. Maybe it is worth to use that instead. @yosefe, wdyt?

Reply:
Will use just ```int``` then.